internal/services/network/network_manager_deployment_resource.go

```diff
@@ -296,8 +306,13 @@ func (r ManagerDeploymentResource) Delete() sdk.ResourceFunc {
 				return fmt.Errorf("deleting %s: %+v", *id, err)
 			}
 
+			deadline, ok := ctx.Deadline()
+			if !ok {
+				return fmt.Errorf("delete context for %s had no deadline", *id)
```

Comment:
These would be internal errors, so for consistency want to be:

```suggestion
				return fmt.Errorf("internal-error: context had no deadline")
```